build.zig

```diff
@@ -482,3 +501,75 @@ fn ensureGitLfsContent(comptime file_path: []const u8) !void {
 inline fn thisDir() []const u8 {
     return comptime std.fs.path.dirname(@src().file) orelse ".";
 }
+
+pub fn linkEmscripten(b: *std.Build, options: Options, exe: *std.Build.CompileStep) !*std.Build.Step.Run {
+    var ems_closure: ?[]const u8 = null;
+    const emsdk_path = b.env_map.get("EMSDK") orelse @panic("Failed to get emscripten SDK path, have you installed & sourced the SDK?");
+    const emscripten_include = b.pathJoin(&.{ emsdk_path, "upstream", "emscripten", "cache", "sysroot", "include" });
+    exe.addSystemIncludePath(emscripten_include);
+    exe.stack_protector = false;
+    exe.disable_stack_probing = true;
+    exe.linkLibC();
+
+    const emlink = b.addSystemCommand(&.{"emcc"});
+    emlink.addArtifactArg(exe);
+    for (exe.link_objects.items) |link_dependency| {
+        switch (link_dependency) {
+            .other_step => |o| emlink.addArtifactArg(o),
+            // .c_source_file => |f| emlink.addFileSourceArg(f.source), // f.args?
+            // .c_source_files => |fs| for (fs.files) |f| emlink.addArg(f), // fs.flags?
+            else => {},
+        }
+    }
+    const out_path = b.pathJoin(&.{ b.pathFromRoot("."), "zig-out", "www", exe.name });
+    std.fs.cwd().makePath(out_path) catch unreachable;
+    const out_file = try std.mem.concat(b.allocator, u8, &.{ "-o", out_path, std.fs.path.sep_str ++ "index.html" });
+    emlink.addArgs(&.{"-sEXPORTED_FUNCTIONS=['_malloc','_emmalloc_realloc','_main']"});
+    //emling.addArgs(&.{"-sLLD_REPORT_UNDEFINED", "-sERROR_ON_UNDEFINED_SYMBOLS=0"});
+    emlink.addArg(out_file);
+    emlink.addArg("-sFILESYSTEM=0");
+    emlink.addArg("-sUSE_GLFW=3");
+    emlink.addArg("-sUSE_WEBGPU=1");
```

Comment:
Library/sample specific flags could be added separately i.e. `linkEmscripten` only has the common stuff